pulpcore/cli/rpm/content.py

```diff
@@ -285,13 +323,152 @@ def upload(
     chunk_size: int,
     **kwargs: t.Any,
 ) -> None:
-    """Create a content unit by uploading a file"""
-    body: t.Dict[str, t.Any]
+    """Create a content unit by uploading a file or files"""
     if isinstance(entity_ctx, PulpRpmPackageContext):
-        result = entity_ctx.upload(file=file, chunk_size=chunk_size, **kwargs)
+        directory = kwargs.get("directory")
+        # Sanity: one of file|directory required
+        if (not (file or directory)) or (file and directory):
+            raise click.ClickException(
+                _("You must specify one (and only one) of --file or --directory.")
+            )
+
+        # Sanity: If directory, repository required
+        final_dest_repo_ctx = kwargs["repository"]
+        if directory and not final_dest_repo_ctx:
+            raise click.ClickException(
+                _("You must specify a --repository to use --directory uploads.")
+            )
+
+        # Sanity: ignore publish|use_temp unless directory has been specified
+        use_tmp = final_dest_repo_ctx and kwargs["use_temp_repository"]
+        do_publish = final_dest_repo_ctx and kwargs["publish"]
+
+        # Sanity: ignore relative_path if directory specified
+        if directory and kwargs["relative_path"]:
+            print(_("relative_path ignored on --directory uploads."))
```

Comment:
Why not fail on this one?
Also i think it should be `click.echo` (don't ask my why...).